sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1672,9 +1707,15 @@ public WriteResult expand(PCollection<T> input) {
 
         // Wrap with a DynamicDestinations class that will provide the proper TimePartitioning.
         if (getJsonTimePartitioning() != null) {
-          dynamicDestinations =
-              new ConstantTimePartitioningDestinations(
-                  dynamicDestinations, getJsonTimePartitioning());
+          if (getJsonClustering() != null) {
```

Comment:
How about adding a single class instead of forking here ?

Reply:
done.